server/aesdsocket.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * Author: Pranav Bharadwaj
+ * Date: 09/30/2022
+ * Formatted for space using codebeautify.org
+ * References:
+ 
+ 1) Getting IPv4 address: https://stackoverflow.com/questions/1276294/getting-ipv4-address-from-a-sockaddr-structure
+ 2) https://beej.us/guide/bgnet/html/#audience
+ 3) https://www.tutorialspoint.com/c_standard_library/c_function_strerror.htm 
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <syslog.h>
+#include <errno.h>
+#include <unistd.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <arpa/inet.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <string.h>
+
+#define LOGFILE_PATH	("/var/tmp/aesdsocketdata")
+#define BUFFER_SIZE	(1024) //chosen as buffer size, can be reduced to 512 as well
+
+//Global to handle signal exit
+bool graceful_exit_handler = false;
+
+//signal handler for SIGINT and SIGTERM
+void signal_handler(int sig) {
+  if (sig == SIGINT || sig == SIGTERM) {
+    syslog(LOG_INFO, "Caught signal. Exiting...");
+    graceful_exit_handler = true;
+    exit(0);
```

Comment:
@prvbharadwaj1731 you won't actually use graceful_exit_handler here with this logic, and you won't be doing cleanup when doing exit() here.